lib/stripe/customer.rb

```diff
@@ -38,25 +38,25 @@ def create_upcoming_invoice(params={}, opts={})
     end
 
     def cancel_subscription(params={}, opts={})
-      response, opts = request(:delete, subscription_url, params, opts)
-      initialize_from({ :subscription => response }, opts, true)
+      self.response, opts = request(:delete, subscription_url, params, opts)
+      initialize_from({ :subscription => response.data }, opts, true)
       subscription
     end
 
     def update_subscription(params={}, opts={})
-      response, opts = request(:post, subscription_url, params, opts)
-      initialize_from({ :subscription => response }, opts, true)
+      self.response, opts = request(:post, subscription_url, params, opts)
+      initialize_from({ :subscription => response.data }, opts, true)
       subscription
     end
 
     def create_subscription(params={}, opts={})
-      response, opts = request(:post, subscriptions_url, params, opts)
-      initialize_from({ :subscription => response }, opts, true)
+      self.response, opts = request(:post, subscriptions_url, params, opts)
+      initialize_from({ :subscription => response.data }, opts, true)
       subscription
     end
 
     def delete_discount
-      _, opts = request(:delete, discount_url)
+      self.response, opts = request(:delete, discount_url)
```

Comment:
Sorry, this one is a little confusing, but we're actually setting `self.response` which is an instance-level variable. So after you've deleted the discount object, you can still get the latest response information on the resource.